Marlin/src/module/configuration_store.cpp

```diff
@@ -494,13 +496,18 @@ void MarlinSettings::postprocess() {
       for (uint8_t q = mesh_num_x * mesh_num_y; q--;) EEPROM_WRITE(dummy);
     #endif // MESH_BED_LEVELING
 
-    _FIELD_TEST(zprobe_zoffset);
-
     #if !HAS_BED_PROBE
       const float zprobe_zoffset = 0;
     #endif
-    EEPROM_WRITE(zprobe_zoffset);
-
+    #if ENABLED(INDEPENDENT_Z_OFFSETS) && ENABLED(DUAL_X_CARRIAGE)
+      _FIELD_TEST(x1_zprobe_zoffset);
+      EEPROM_WRITE(x1_zprobe_zoffset);
+      _FIELD_TEST(x2_zprobe_zoffset);
```

Comment:
It's overkill to have a `_FIELD_TEST` for every field. A single instance would be fine here.